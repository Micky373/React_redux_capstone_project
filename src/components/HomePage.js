import { useSelector } from 'react-redux';
import DataRender from './DataRender';

const HomePage = () => {
  const countries = useSelector((state) => state);
  const loadingStatus = countries.loading;
  return (
    <div>
      <div className="nav_bar">
        <span className="nav_header">Top 20 African countries COVID cases</span>
        <i className="fa fa-microphone microphone_button" aria-hidden="true" />
        <i className="fa fa-gear nav_buttons" aria-hidden="true" />
      </div>
      <div className="image_wrapper">
        <div className="image_container" />
      </div>
      <div className="country_link_container">
        {loadingStatus === false
          ? countries.data.map(
            (country) => (
              <>
                <DataRender
                  country={country.CountryName}
                  key={country.ID}
                  total={country.TotalConfirmed}
                />
              </>
            ),
          ) : <h1>***Loading***</h1>}
      </div>
    </div>
  );
};

export default HomePage;
