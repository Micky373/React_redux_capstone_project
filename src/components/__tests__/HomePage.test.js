import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import HomePage from '../HomePage';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

const element = (
  <Provider store={store}>
    <HomePage />
  </Provider>
);

it('it works', () => {
  const tree = renderer.create(element).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Render check', () => {
  render(element);
  const dataRender = screen.getByText('***Loading***');
  expect(dataRender).toBeInTheDocument();
});
