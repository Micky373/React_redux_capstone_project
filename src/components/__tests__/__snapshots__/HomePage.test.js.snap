// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
<div>
  <div
    className="nav_bar"
  >
    <span
      className="nav_header"
    >
      Top 20 African countries COVID cases
    </span>
    <i
      aria-hidden="true"
      className="fa fa-microphone microphone_button"
    />
    <i
      aria-hidden="true"
      className="fa fa-gear nav_buttons"
    />
  </div>
  <div
    className="image_wrapper"
  >
    <div
      className="image_container"
    />
  </div>
  <p
    className="search_item_container"
  >
    Search:  
    <input
      onChange={[Function]}
      placeholder="Search by country"
      type="text"
      value=""
    />
  </p>
  <div
    className="country_link_container"
  >
    <h1>
      ***Loading***
    </h1>
  </div>
</div>
`;
